### PR TITLE
Zero exit code if no tests executed (fixes #636)

### LINKED ISSUE #636: Paratest fails when no tests are found
| Q | A
| --- | ---
| ParaTest version | 6.3.1
| PHPUnit version | 9.5.10
| PHP version | 8.0.10

#### Summary

When the configured directories don't contain any tests, Paratest behaves differently than PHPUnit.

#### Current behavior

PHPUnit:

```
PHPUnit 9.5.10 by Sebastian Bergmann and contributors.

No tests executed!

```

Paratest:

```
In SuiteLoader.php line 172:

  No path or configuration provided (tests must end with Test.php)

```

#### How to reproduce: command, code and error stack trace

Delete all tests and try running phpunit and paratest to see the difference.

#### Expected behavior

Same as PHPUnit, Paratest should succeed with a note that there are no tests.

### PR DESCRIPTION
**Pull request by remorhaz**
Fixes #636.

### CODE REVIEW

File: src/Runners/PHPUnit/BaseRunner.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ abstract class BaseRunner implements RunnerInterface
      *
      * @var int
      */
-    protected $exitcode = -1;
+    protected $exitcode = 0;
```
Comment (remorhaz) [inline src/Runners/PHPUnit/BaseRunner.php:48]: Now default (without running any test) exit code is 0.

File: src/Runners/PHPUnit/SuiteLoader.php [hunk 2]
```diff
@@ -126,7 +127,11 @@ public function load(): void
     {
         $this->loadConfiguration();
 
+        $testSuiteCollection = null;
         if (($path = $this->options->path()) !== null) {
+            if (realpath($path) === false) {
```
Comment (remorhaz) [inline src/Runners/PHPUnit/SuiteLoader.php:132]: PHPUnit fails if provided path is not an existing file or directory, and we implement same behavior.

File: src/Runners/PHPUnit/SuiteLoader.php [hunk 3]
```diff
@@ -136,39 +141,29 @@ public function load(): void
...
             && ! $this->configuration->testSuite()->isEmpty()
         ) {
-            $testSuiteCollection = $this->configuration->testSuite()->asArray();
-            if (count($this->options->testsuite()) > 0) {
-                $suitesName = array_map(static function (TestSuite $testSuite): string {
-                    return $testSuite->name();
-                }, $testSuiteCollection);
-                foreach ($this->options->testsuite() as $testSuiteName) {
-                    if (! in_array($testSuiteName, $suitesName, true)) {
-                        throw new RuntimeException("Suite path {$testSuiteName} could not be found");
```
Comment (remorhaz) [inline src/Runners/PHPUnit/SuiteLoader.php:153]: PHPUnit returns success on running non-existent suite; we implement same behavior and remove this check.

File: src/Runners/PHPUnit/SuiteLoader.php [hunk 4]
```diff
@@ -136,39 +141,29 @@ public function load(): void
...
-            }
+            );
 
             foreach ($testSuiteCollection as $testSuite) {
                 $this->loadFilesFromTestSuite($testSuite);
             }
         }
 
-        if (count($this->files) === 0 && ! is_array($this->suitesName)) {
+        if ($path === null && $testSuiteCollection === null) {
```
Comment (remorhaz) [inline src/Runners/PHPUnit/SuiteLoader.php:166]: We fail only on invalid configuration now.

**Review by Slamdunk — APPROVED** [review]

**Comment by Slamdunk** [comment]
Thank you @remorhaz

**Pull request merged**